qutebrowser/javascript/webengine_caret.js

```diff
@@ -0,0 +1,1401 @@
+/* eslint-disable max-lines, max-len, max-statements, complexity,
+max-params, default-case, valid-jsdoc */
+/**
+ * Copyright 2018 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+ *
+ * This file is part of qutebrowser.
+ *
+ * qutebrowser is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * qutebrowser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+ */
```

Comment:
The original file states:

```
/* Copyright (c) 2014 The Chromium Authors. All rights reserved.
 * Use of this source code is governed by a BSD-style license that can be
 * found in the LICENSE file. */
```

We should probably add their license here to be on the safe side, as it's largely their code. Instead of just adding those lines and having a second `LICENSE` file in the repos, I'd suggest simply copying their entire [license file](https://chromium.googlesource.com/chromium/src/+/master/LICENSE) (but with the copyright date adjusted to 2014) and adding it as a comment here.